documentation/bolt_examples.md

```diff
@@ -123,3 +123,226 @@ To convert the `restart_service.ps1` script to a task, giving you the ability to
+
+Bolt uses a `Puppetfile` to install module content from the Forge. A `Puppetfile` is a formatted text file that specifies the modules and data you want in each environment.
+
+1. Create a file named `Puppetfile` in the project directory — which describes the Forge content you want to install: 
+
+```     
+# Modules from the Puppet Forge
+mod 'puppetlabs-chocolatey', '4.1.0'
+## dependencies
+mod 'puppetlabs-stdlib', '4.13.1' #install latest
+ mod 'puppetlabs-powershell', '2.3.0'
+mod 'puppetlabs-registry', '2.1.0'
+        
+# Modules from Git
+# Examples: https://github.com/puppetlabs/r10k/blob/master/doc/puppetfile.mkd#examples
+#mod 'apache',
+#  :git    => 'https://github.com/puppetlabs/puppetlabs-apache',
+#  :commit => 'de290646f97e04b4b8e42c70f6e01e860c394ce7'     
+ ```
+
+2. Add the Puppetfile to a version control service, for example, Github.
+
+3. Add the following code to the Puppetfile — with your git user account: 
+
+```     
+mod 'puppet_choco_tap',
+    :git    =>  'https://github.com/<your git user>/puppet_choco_tap.git'
+```
+
+4. From inside the project directory, install the Forge content by running: 
+
+```
+bolt puppetfile install
+```
+
+After it runs, you can see a `modules` directory inside the project directory, containing the modules you specified in the `Puppetfile`.
+
+### 3. Write a Bolt plan to apply Puppet code and orchestrate the deployment of a package resource using the Chocolatey provider. Plans allow you to run more than one task with a single command.
+
+1. Inside your project directory, in a file called `/plan/installer.pp`, create a plan called `puppet_chocolatey_tap::installer`. 
+
+The folder tree should look like this:
+
+```
+├──plan
+   └── installer.pp 
+```
+
+2. Copy the following code into the `installer.pp` file:
+
+```
+plan puppet_choco_tap::installer(
+   TargetSpec                                   $nodes,
+   String                                       $package,
+   Variant[Enum['absent', 'present'], String ]  $ensure = 'present',
+ ){
+
+ apply_prep($nodes)
+
+apply($nodes){
+  include chocolatey
+ }
+apply($nodes){
+  package { $package :
+    ensure    => $ensure,
+    provider  => 'chocolatey',
+    }
+  }
+}
+```
+
+Take note of the following features of the plan:
+
+- It has three parameters: the `TargetSpec`, a `package` string for the package name, and the `ensure` state of the package which allows for version, absent or present.
+- It has the `apply_prep` function call, which is used to install modules needed by `apply` on remote nodes as well as to gather facts about the nodes.
+- The first `apply` block installs the Chocolatey package manager. The Chocolatey provider is also deployed as a library with the Puppet agent in `apply_prep`.
+- The second `apply` block installs a specified package using the Chocolatey provider.
+
+2. To verify that the `puppet_chocolatey_tap` plan is available, run the following command: 
```

Comment:
With the workflow described here, you would have to first check in the code to your remote repository, then re-run the puppetfile install. I would recommend though, describing checking in the code here and adding the line to the puppetfile here. As written the plan will not be in the correct location because r10k (bolt puppetfile install) has not installed the module under `modules`.

Reply:
@donoghuc I've made all the other changes you've suggested, thank you!
I'm not too clear on what you mean here though. Could you please at this step?

Reply:
So to me it looks like the way module content ends up in the expected location is for r10k to fetch the code from the remote repo and add it under 'modules/puppetlabs/plans'. So any change you want to make you must push to remote and then re-install with `bolt puppetfile install`. I think that we should explicitly describe this model if that is how the author wants people to use this workflow. In addiition to that I think we need to have the instructions explicitly add the steps for checking in the newly minted plan into the remote repo and then pulling the change with r10k (bolt puppetfile install). Otherwise this will not work as expected bc the plan will not be found.

Reply:
@donoghuc Let me know how this looks, thanks!